lte/gateway/c/oai/tasks/nas/emm/NasTransportHdl.c

```diff
@@ -113,12 +113,16 @@ int emm_proc_uplink_nas_transport(mme_ue_s1ap_id_t ue_id, bstring nas_msg_pP) {
   if (emm_ctxt_p != NULL) {
     ue_mm_context_t* ue_mm_context_p =
         PARENT_STRUCT(emm_ctxt_p, struct ue_mm_context_s, emm_context);
-    /* check if the non EPS service control is enable and combined attach*/
+    /* check if the non EPS service control is enable and combined attach. If
+     * in SMS_ORC8R, we still want to send the uplink message, but we should
+     * disable vlr checks since SGs is not present.*/
     if (((_esm_data.conf.features & MME_API_SMS_SUPPORTED) ||
-         (_esm_data.conf.features & MME_API_CSFB_SMS_SUPPORTED)) &&
+         (_esm_data.conf.features & MME_API_CSFB_SMS_SUPPORTED ||
+	  _esm_data.conf.features & MME_API_SMS_ORC8R_SUPPORTED)) &&
         (emm_ctxt_p->attach_type == EMM_ATTACH_TYPE_COMBINED_EPS_IMSI)) {
-      // check if vlr reliable flag is true for sgs association
-      if (mme_ue_context_get_ue_sgs_vlr_reliable(ue_id) == true) {
+      // check if vlr reliable flag is true for sgs association.
+      if (mme_ue_context_get_ue_sgs_vlr_reliable(ue_id) == true ||
+	  _esm_data.conf.features & MME_API_SMS_ORC8R_SUPPORTED) {
```

Comment:
For better readability:
`if ( mme_ue_context_get_ue_sgs_vlr_reliable(ue_id) ||`
	 `(_esm_data.conf.features & MME_API_SMS_ORC8R_SUPPORTED) ) {`